language/def_spec.rb

```diff
@@ -630,5 +630,77 @@ def some_method; end
 
       DefSpecsLambdaVisibility.should have_private_instance_method("some_method")
     end
+    
+    it "creates instance methods using instance_eval on an anon class" do
+      m = Class.new
+      m.instance_eval do
+        def klass_method
+          :test
+        end
+      end
+      
+      m.klass_method.should == :test
+    end
+    
+    it "creates instance methods using instance_exec on an anon class" do
+      m = Class.new
+      m.instance_exec(1) do |param|
+        @stuff = param
+        
+        def klass_method
+          @stuff
+        end
+      end
+      
+      m.klass_method.should == 1
+    end
+    
+    it "creates instance methods using instance_eval inside an anon class" do
+      m = Class.new do
+        instance_eval do
+          def klass_method
+            :test
+          end          
+        end
+      end
+      
+      m.klass_method.should == :test
+    end
+    
+    it "creates instance methods using instance_exec inside an anon class" do
+      m = Class.new do
+        instance_exec(2) do |param|
+          @stuff = param
+          
+          def klass_method
+            @stuff
+          end          
+        end
+      end
+      
+      m.klass_method.should == 2
+    end
+    
+    it "creates instance methods using instance_eval on a named class" do
+      DefSpecIval::InstanceEval.instance_eval do
+        def instance_eval_method
+          :test
+        end
+      end
+      
+      DefSpecIval::InstanceEval.instance_eval_method.should == :test
+    end
+    
+    it "creates instance methods using instance_exec on a named class" do
+      DefSpecIval::InstanceEval.instance_exec(3) do |param|
+        @stuff = param
+        
+        def instance_exec_method
+          @stuff
+        end
+      end
+      
+      DefSpecIval::InstanceEval.instance_exec_method.should == 3
+    end 
```

Comment:
What was the real issue here for Opal? the method got defined on the wrong module?
The specs around line 518 should already cover the cases inside instance_eval.